tools/wake-format/emitter.cpp

```diff
@@ -418,15 +416,26 @@ wcl::doc Emitter::walk_ideq(ctx_t ctx, CSTElement node) { return walk_placeholde
 wcl::doc Emitter::walk_if(ctx_t ctx, CSTElement node) { return walk_placeholder(ctx, node); }
 
 wcl::doc Emitter::walk_import(ctx_t ctx, CSTElement node) {
-  ASSERT_TOKEN(node, CST_IMPORT);
-  // return formatter()
-  //     .token(TOKEN_KW_FROM)
-  //     .ws()
-  //     .walk(CST_ID, [this](ctx_t ctx, CSTElement node) { return walk_identifier(ctx, node); })
-  //     .ws()
-  //     .token(TOKEN_KW_IMPORT)
-  //     .ws()
-  return walk_placeholder(ctx, node);
+  assert(node.id() == CST_IMPORT);
+  return formatter()
+      .token(TOKEN_KW_FROM)
+      .ws()
+      .walk(CST_ID, WALK(walk_identifier))
+      .ws()
+      .token(TOKEN_KW_IMPORT)
+      .ws()
+      .fmt_if(CST_KIND, formatter().walk(CST_KIND, WALK(walk_kind)).ws())
+      .fmt_if(CST_ARITY, formatter().walk(CST_ARITY, WALK(walk_arity)).ws())
+      // clang-format off
+      .fmt_if_else(
+          TOKEN_P_HOLE,
+          formatter().walk(TOKEN_P_HOLE, WALK(walk_token)),
```

Comment:
not something to do here but I wonder if it would make more sense to make the WALK macro into like `WALK(TOKEN_P_HOLE, walk_token)` and you could use it in the fluent API as like a macro method?

Reply:
I actually prefer this, but `WALK({CST_ID, CST_APP}, walk_node)` doesn't seem to work :(

Reply:
Well you'd have to change the macro but it should work

```
#define NEW_WALK(pred, method) walk(pred, WALK(method))
```

Reply:
The issue is the `,` `{CST_ID, CST_APP}`. It messes up the expansion.

There might be a way to escape the comma, that that would be a weird public api

Reply:
oh shit I didn't think about that, what if we flip it around and we make the second argument a vararg and then we expand it?

Reply:
TBH I'm partial to having it be the first parameter. Lets hold on this change for a bit and feel it out while implementing subtrees